modules/openapi-generator/src/main/resources/Java/libraries/rest-assured/api.mustache

```diff
@@ -139,8 +145,11 @@ public class {{classname}} {
          * @return {{returnType}}
          */
         public {{{returnType}}} executeAs(Function<Response, Response> handler) {
-            Type type = new TypeToken<{{{returnType}}}>(){}.getType();
-            return execute(handler).as(type);
+            {{#gson}}Type type = new TypeToken<{{{returnType}}}>(){}.getType();
+            {{/gson}}{{#jackson}}{{#isListContainer}}Type type = TypeFactory.defaultInstance().constructCollectionLikeType(java.util.ArrayList.class, {{{returnBaseType}}}.class);
+            {{/isListContainer}}{{#isMapContainer}}Type type = TypeFactory.defaultInstance().constructMapLikeType(java.util.LinkedHashMap.class, String.class, {{{returnBaseType}}}.class);
```

Comment:
Could you explain, why you don't use mapping on Type, like in Gson? Sounds like it is possible https://www.baeldung.com/jackson-object-mapper-tutorial

Reply:
`TypeToken` is `com.google.gson.reflect.TypeToken`, so `gson` specific.

I thought `TypeFactory` is the way to go, but from your article it seems that `com.fasterxml.jackson.core.type.TypeReference` would be similar. Let me try this.

Reply:
Jackson `TypeReference` has no bridge to `java.lang.reflect.Type` like gson has, so this does not work with rest-assured.

But since rest-assured is wrapping all the serialization library, the way to go might be to use [`TypeRef`](https://github.com/rest-assured/rest-assured/blob/master/rest-assured-common/src/main/java/io/restassured/common/mapper/TypeRef.java).